### PR TITLE
Cleanup artifacts

### PR DESCRIPTION
**Pull request by luca-heltai**
We are having some issues with the artifacts not uploading. This checks for all artifacts older than a week (every night) and removes them. It should fix our currently failing PRs.

### CODE REVIEW

**Review by peterrum — APPROVED** [review]
Looks reasonable. But you could also set the life time in settings `Actions->General->Artifact and log retention`

**Comment by luca-heltai** [comment]
I personally do not have access to those options. I think only the owner of the dealii github repository has.

**Comment by peterrum** [comment]
@bangerth @tamiko @tjhei I guess you (as owners of the dealii organization) are the only ones who could make the change in the settings. Else let's go with @luca-heltai's solution!

**Comment by tjhei** [comment]
I have access to that option. It is set to 90 days right now. What setting makes sense?

File: .github/workflows/cleanup.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    - cron: '0 1 * * *' # every night at 1 am UTC
+
+jobs:
+  delete-artifacts:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: kolpav/purge-artifacts-action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          expire-in: 7days # Set this to 0 to delete all artifacts
```
Comment (peterrum) [inline .github/workflows/cleanup.yml:13]: @tjhei?

Comment (tjhei) [reply .github/workflows/cleanup.yml:13]: why 7 days and not something like 14?

Comment (peterrum) [reply .github/workflows/cleanup.yml:13]: I guess everything below 90 is fine. So why not 14 ;) Out of curiosity: how much volume are the artifact consuming right now? You can check that if you go to the organization and go to `Billing and plans`.

Comment (luca-heltai) [reply .github/workflows/cleanup.yml:13]: 90 seems too much right now. I think we have reached the maximum allowable space with our windows artifacts (these are roughly 400MB for each successfull build). I would have thought that 7 days was enough, but if you think we need more, let's do 14.

Comment (peterrum) [reply .github/workflows/cleanup.yml:13]: > with our windows artifacts (these are roughly 400MB for each successfull build)

Wait a second. What is the purpose of storing that as artifact!?

Comment (peterrum) [reply .github/workflows/cleanup.yml:13]: > I guess everything below 90 is fine. So why not 14 ;) Out of curiosity: how much volume are the artifact consuming right now? You can check that if you go to the organization and go to Billing and plans.

@tjhei  Now, I am curious how much data has been accumulated. Terabytes?

**Comment by tjhei** [comment]
I have no idea where I can see how much space we use. I changed to 30 days for now. Let's see what happens...

File: .github/workflows/cleanup.yml (hunk 1 above)
Comment (luca-heltai) [reply .github/workflows/cleanup.yml:13]: The windows artifact is a full working installation of deal.II, that can be used as a binary installation package. Pretty much like our docker images of the master branch, but for windows. These are used by a few guys around the globe to keep their windows apps up to date.

**Comment by luca-heltai** [comment]
Since this seems to have been resolved by freeing up space, I'm closing the PR.

**Pull request closed without merging**